task-ballerina/src/task/commons.bal

```diff
@@ -54,3 +108,41 @@ public type AppointmentData record {|
     string daysOfWeek?;
     string year?;
 |};
+
+# Possible types of parameters that can be passed into the `Policy`.
+public type MisfirePolicy TimerTaskPolicy|AppointmentTaskPolicy;
```

Comment:
If we have two different `Misfire Policies` for `TimerConfiguration` and the `AppointmentConfiguration` why do we  need to unify them and put it inside both the `TimerConfiguration` and the `AppointmentConfiguration`? It will allow invalid values to be passed to the configurations, and we have to have additional validations.

We can just use the `TimerTaskPolicy` inside the `TimeConfiguration` and the `AppointmentTaskPolicy` inside the `AppointmentConfiguration`.

Also, I don't think `task` keyword is needed for the type name in `TimerTaskPolicy` and the `AppointmentTaskPolicy`. Something like `TimerMisfirePolicy` would be better.